pkg/logql/log/parser.go

```diff
@@ -336,52 +340,41 @@ func (u *UnpackParser) Process(line []byte, lbs *LabelsBuilder) ([]byte, bool) {
 		return line, true
 	}
 	u.lbsBuffer = u.lbsBuffer[:0]
-	it := jsoniter.ConfigFastest.BorrowIterator(line)
-	defer jsoniter.ConfigFastest.ReturnIterator(it)
-
-	entry, err := u.unpack(it, line, lbs)
+	entry, err := u.unpack(line, lbs)
 	if err != nil {
 		lbs.SetErr(errJSON)
 		return line, true
 	}
 	return entry, true
 }
 
-func (u *UnpackParser) unpack(it *jsoniter.Iterator, entry []byte, lbs *LabelsBuilder) ([]byte, error) {
-	// we only care about object and values.
-	if nextType := it.WhatIsNext(); nextType != jsoniter.ObjectValue {
-		return nil, fmt.Errorf("expecting json object(%d), got %d", jsoniter.ObjectValue, nextType)
-	}
+func (u *UnpackParser) unpack(entry []byte, lbs *LabelsBuilder) ([]byte, error) {
 	var isPacked bool
-	_ = it.ReadMapCB(func(iter *jsoniter.Iterator, field string) bool {
-		switch iter.WhatIsNext() {
-		case jsoniter.StringValue:
-			// we only unpack map[string]string. Anything else is skipped.
-			if field == PackedEntryKey {
-				s := iter.ReadString()
-				// todo(ctovena): we should just reslice the original line since the property is contiguous
-				// but jsoniter doesn't allow us to do this right now.
-				// https://github.com/buger/jsonparser might do a better job at this.
-				entry = append(entry[:0], []byte(s)...)
-				isPacked = true
-				return true
-			}
-			if !lbs.ParserLabelHints().ShouldExtract(field) {
-				iter.Skip()
-				return true
-			}
-			if lbs.BaseHas(field) {
-				field = field + duplicateSuffix
-			}
-			// append to the buffer of labels
-			u.lbsBuffer = append(u.lbsBuffer, field, iter.ReadString())
-		default:
-			iter.Skip()
+
+	err := jsonparser.ObjectEach(entry, func(key []byte, value []byte, dataType jsonparser.ValueType, offset int) error {
+		if dataType != jsonparser.String {
+			return nil
 		}
-		return true
+		if bytes.Equal(key, []byte(PackedEntryKey)) {
+			entry = bytes.ReplaceAll(value, []byte("\\\""), []byte("\""))
```

Comment:
Not sure this is correct. I think I'm just pleasing the test here with this @slim-bean ?